platforms/jvm/plugins-application/src/main/java/org/gradle/api/internal/plugins/StartScriptTemplateBindingFactory.java

```diff
@@ -188,7 +237,25 @@ String createJoinedAppHomeRelativePath(String scriptRelPath) {
             appHomeRelativePath.add("..");
         }
 
-        return Joiner.on(windows ? "\\" : "/").join(appHomeRelativePath);
+        return Joiner.on(getPathElementSeparator()).join(appHomeRelativePath);
+    }
+
+    /**
+     * The separator used to separate each element in a path.
+     *
+     * @return the path element separator
+     */
+    private String getPathElementSeparator() {
+        return windows ? "\\" : "/";
+    }
+
+    /**
+     * The separator used to separate each path in a multi-path argument.
+     *
+     * @return the multi-path separator
+     */
+    private String getMultiPathSeparator() {
+        return windows ? ";" : ":";
```

Comment:
I think you can use `File.pathSeparator`

Reply:
See above.